src/core/src/main/java/org/locationtech/geogig/model/RevFeatureBuilder.java

```diff
@@ -71,10 +76,33 @@ public RevFeatureBuilder addProperty(Property featureProp) {
      */
     public RevFeatureBuilder addValue(@Nullable Object value) {
         value = safeCopy(value);
+        if (value instanceof Geometry) {
+            value = normalizeIfNeeded((Geometry) value);
+        }
         this.values.add(value);
         return this;
     }
 
+    Geometry normalizeIfNeeded(Geometry value) {
+        if (value instanceof Polygon) {
+            value.normalize();
+        } else if (value instanceof MultiPolygon
```

Comment:
Since MultiPolygon is a subclass of GeometryCollection, could we condense this check to be something like:
```java
else if (GeometryCollection.class.isAssignableFrom(value.getClass())) {
    normalize(GeometryCollection.class.cast(value));
}
```
It's more or less the same thing, but avoids `instanceof` for MultiPolygon or class checking for GeometryCollection objects. Technically though, what you have will only call `normalize` if the Geometry `value` is a MultiPolygon (or subclass) or a GeometryCollection specifically, where as my proposed way would also include any other subclasses of GeometryCollection.